core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -241,12 +246,17 @@ public RemoteLogManager(RemoteLogManagerConfig rlmConfig,
 
         indexCache = new RemoteIndexCache(rlmConfig.remoteLogIndexFileCacheTotalSizeBytes(), remoteLogStorageManager, logDir);
         delayInMs = rlmConfig.remoteLogManagerTaskIntervalMs();
-        rlmScheduledThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize());
+        rlmCopyThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerCopierThreadPoolSize(),
+            "RLMCopyThreadPool", "kafka-rlm-copy-thread-pool-");
+        rlmExpirationThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerExpirationThreadPoolSize(),
+            "RLMExpirationThreadPool", "kafka-rlm-expiration-thread-pool-");
+        followerThreadPool = new RLMScheduledThreadPool(rlmConfig.remoteLogManagerThreadPoolSize(),
+            "RLMFollowerScheduledThreadPool", "kafka-rlm-follower-thread-pool-");
 
         metricsGroup.newGauge(REMOTE_LOG_MANAGER_TASKS_AVG_IDLE_PERCENT_METRIC, new Gauge<Double>() {
             @Override
             public Double value() {
-                return rlmScheduledThreadPool.getIdlePercent();
+                return rlmCopyThreadPool.getIdlePercent();
```

Comment:
This is a miss on my side - I am happy to circle back to the KIP to mention that this metric will either a) no longer show a reasonable value or b) will show the value of the copy pool only. Also another miss on my side is that we should surface the same metrics per thread pool that we have now for the new thread pools. But I am happy to pick this up once this PR is closed 😊

Reply:
@clolov can we also update the KIP to have a separate config for the follower threadpool. We can then use that one instead of _remoteLogManagerThreadPoolSize_